Promtail: Add new target for receiving GCP PubSub Push messages (#6777)
grafana/loki: docs/sources/clients/promtail/scraping.md

see the [configuration](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#windows_events) section for more information.

## GCP Log scraping

> Promtail supports scraping cloud resource logs (say GCS bucket logs, Load Balancer logs, Kubernetes Cluster logs) from GCP.
> Configs are set in `gcplog` section in `scrape_config`.

Reviewer: Small suggestions here:
```suggestion
Promtail supports scraping cloud resource logs such as GCS bucket logs, load balancer logs, and Kubernetes cluster logs from GCP.
Configuration is specified in the `gcplog` section, within `scrape_config`.
```

  - `__gcp_resource_labels_<NAME>`
    In the example above, the `project_id` label from a GCP resource was transformed into a label called `project` through `relabel_configs`.

### Push

```yaml
  - job_name: gcplog
    gcplog:
      subscription_type: "push"
      use_incoming_timestamp: false
      labels:
        job: "gcplog-push"
      server:
        http_listen_address: 0.0.0.0
        http_listen_port: 8080
    relabel_configs:
      - source_labels: ['__gcp_message_id']
        target_label: 'message_id'
      - source_labels: ['__gcp_attributes_logging_googleapis_com_timestamp']
        target_label: 'incoming_ts'
```

> When configuring the GCP Log push target, Promtail will start a HTTP server listening in port `8080` as configured in the `server`
> section. This server will expose just one endpoint `POST /gcp/api/v1/push` responsible for receiving logs from GCP.
> 
> It also supports `relabeling` and `pipeline` stages just like other targets.

Reviewer: Some small grammar improvements here.
```suggestion
When configuring the GCP Log push target, Promtail will start an HTTP server listening on port `8080`, as configured in the `server`
section. This server exposes the single endpoint `POST /gcp/api/v1/push`, responsible for receiving logs from GCP.

It also supports `relabeling` and `pipeline` stages.
```
